server/src/game/objects/obstacle.ts

```diff
@@ -307,6 +307,19 @@ export class Obstacle extends BaseGameObject {
             v2.set(lootPos, v2.add(this.pos, v2.rotate(def.lootSpawn.offset, this.rot)));
         }
 
+        if (
+            params.source instanceof Player &&
+            params.source.hasPerk("scavenger") &&
+            def.loot.length != 0
```

Comment:
if player had scavenger obstacles that have loot by default would still drop an extra item